Recast this as a node/baker/DAL tutorial (#495)
trilitech/tezos-developer-docs: docs/tutorials/join-dal-baker/run-baker.md

title: "Step 4: Run an Octez baking daemon"
authors: Tezos core developers, Tim McMackin
last_update:
  date: 19 December 2024
---

Now that you have a layer 1 node and a DAL node, you can run a baking daemon that can create blocks and attests to DAL data.
If you already have a baking daemon, you can restart it to connect to the DAL node.

1. Optional: Set up a remote signer to secure the keys that the baker uses as described in [Signer](https://tezos.gitlab.io/user/key-management.html#signer) in the Octez documentation.

1. To run a baking daemon that connects to the DAL, start it as usual and pass the URL to your DAL node to it with the `--dal-node` argument:

   ```bash
   octez-baker-PsParisC run with local node "$HOME/.tezos-node" my_baker --liquidity-baking-toggle-vote pass --adaptive-issuance-vote on --dal-node http://127.0.0.1:10732
   ```

   Note that the command for the baker depends on the protocol version.

>    This example uses the ParisC protocol, so the command starts with `octez-baker-PsParisC`.

Reviewer: Is there a way to use a variable here for the protocol suffix that gets instantiated by Docusaurus at build time, to avoid changing the tutorial  at a protocol shift?

Author: There are a few ways to do this -- I'll add a note as part of our discussion about a list of to-dos for each protocol upgrade.

Look up how to run programs persistently in the documentation for your operating system.
You can also refer to [Run a persistent baking node](https://opentezos.com/node-baking/baking/persistent-baker/) on opentezos.com.

   For example, if your operating system uses the `systemd` software suite, your service file might look like this example:

   ```systemd
   [Unit]
   Description=Octez baker

>    Wants = network-online.target

Reviewer: why are certain assignments spaced and others are not? Couldn't we be consistent, for instance, never spaced?

Author: Made the spacing consistent.

- Attach DAL attestations: `ready to attach DAL attestation ...`

Whether these messages appear or not after the attestation delay, proceed to [Step 5: Verifications](/tutorials/join-dal-baker/verify-rights).

## Optional: Run an accuser

The accuser is a daemon that monitors blocks and looks for problems, such as bakers who double-sign blocks or inject multiple attestations.

> If it finds a problem, it posts a denunciation operation, which penalizes the baker.

Reviewer: ```suggestion
If it finds a problem, it posts a denunciation operation, which leads to penalizing the misbehaving baker.
```

> You don't have to run an accuser, but if you do, you can receive some of the penalties when the accuser denounces a baker.

Reviewer: ```suggestion
You don't have to run an accuser, but if you do, you can receive as a reward part of the penalties of the denounced baker.
```
